helix-term/src/commands.rs

```diff
@@ -1076,16 +1077,22 @@ mod cmd {
             .language_config()
             .map(|config| config.auto_format)
             .unwrap_or_default();
-        if autofmt {
-            doc.format(view.id); // TODO: merge into save
-        }
-        helix_lsp::block_on(tokio::spawn(doc.save()));
+        let fmt = if autofmt {
+            doc.format().map(|fmt| {
```

Comment:
Can we get `format()` to return `Future` without `Option` since that only checks for language server.

Maybe `autofmt` should take it into account whether language server is available so we don't have to do the first branch or can just unwrap?

Reply:
For the return type of `format()`, I can see these options:
1) `Option<impl Future>`
2) `impl Future`, and panic if the language server isn't there
3) `impl Future`, and return a boxed future with no changes if the language server isn't there.

My preference is for (1).

One alternative that would make this usage nicer would be to add a `Document::auto_format` method that's the same as `Document::format` but returns `None` if autoformatting is off.

Reply:
I think returning `Option` is fine, it's what I've done elsewhere. Returning a future means we need to box values, it's an unnecessary cost if we're just trying to signal no formatting can be done.

Reply:
I tried adding a `Document::auto_format` method. See what you think...